### PR TITLE
Zeta ABI

### PR DESCRIPTION
**Pull request by konrads**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -108,4 +113,12 @@ mod zeta {
     pub(crate) fn liquidate(ctx: Context<Liquidate>, size: u64) -> Result<()> {
         Ok(())
     }
+
+    pub fn close_open_orders(ctx: Context<CloseOpenOrders>, _map_nonce: u8) -> Result<()> {
```
Comment (xjcaa) [inline src/lib.rs]: should this also be (crate)

File: src/account.rs [hunk 2]
```diff
@@ -0,0 +1,589 @@
...
+    pub expiry_series: [ExpirySeries; 2],         // 32 * 6 = 192
+    pub expiry_series_padding: [ExpirySeries; 4], //
+    pub total_insurance_vault_deposits: u64,      // 8
+    pub asset: Asset,                             // 1
+    pub expiry_interval_seconds: u32,             // 4
+    pub new_expiry_threshold_seconds: u32,        // 4
+    pub perp_parameters: PerpParameters,          // 24
+    pub perp_sync_queue: Pubkey,                  // 32
+    pub oracle_backup_feed: Pubkey,               // 32
+    pub padding: [u8; 966],                       // 966
```
Comment (xjcaa) [inline src/account.rs]: Could this be updated to the latest - it's slightly out of date, theres a few fields like 

perpOnly, flexUnderlying

File: src/account.rs [hunk 3]
```diff
@@ -0,0 +1,589 @@
...
+    ImmediateOrCancel = 3,
+    PostOnlySlide = 4,
+}
+
+#[repr(u8)]
+#[derive(AnchorSerialize, AnchorDeserialize, PartialEq, Eq, Clone, Copy)]
+pub enum Asset {
+    SOL = 0,
+    BTC = 1,
+    ETH = 2,
```
Comment (xjcaa) [inline src/account.rs:556]: Theres one more APT

File: src/account.rs [hunk 4]
```diff
@@ -0,0 +1,589 @@
...
+        // There shouldn't be a case where you set a strike without resetting it first.
+        if self.is_set() {
+            return Err(error!(ZetaError::CannotSetInitializedStrike));
+        }
+        self.is_set = true;
+        self.value = strike;
+        Ok(())
+    }
+
+    pub fn reset(&mut self) -> Result<()> {
```
Comment (xjcaa) [inline src/account.rs]: set, reset, don't need to be in here, along with the errors

File: src/account.rs [hunk 5]
```diff
@@ -0,0 +1,589 @@
...
+
+        // Perps have a different layout
+        let perp_margin_requirement: u64 = self.perp_product_ledger.get_initial_margin(
+            spot,
+            &zeta_group.perp,
+            spot,
+            &zeta_group.margin_parameters,
+        );
+
+        msg!(
```
Comment (xjcaa) [inline src/account.rs]: can probably remove this log

**Review by xjcaa — APPROVED** [review]
Looks good just the comments i've left which should be easy to fix

could you add to changelog

File: src/lib.rs (hunk 1 above)
Comment (konrads) [reply src/lib.rs]: yes

File: src/account.rs (hunk 2 above)
Comment (konrads) [reply src/account.rs]: cheers

File: src/account.rs (hunk 3 above)
Comment (konrads) [reply src/account.rs:556]: cheers

File: src/account.rs (hunk 5 above)
Comment (konrads) [reply src/account.rs]: k

**Pull request merged**